tests/unit/test_kernel.py

```diff
@@ -76,21 +64,379 @@ def test_approximator_valid(self) -> None:
         kernel.approximate_kernel_matrix_row_sum_mean(x=x, approximator=approximator)
         approximator_approximate_method.assert_called_once_with(x)
 
+    def test_update_kernel_matrix_row_sum_zero_max_size(self) -> None:
+        """
+        Test how the method update_kernel_matrix_row_sum handles a zero max_size.
+        """
+        # Define parameters for data
+        length_scale = 1 / np.sqrt(2)
+        x = jnp.array([0.0, 1.0, 2.0, 3.0, 4.0]).reshape(-1, 1)
+
+        # Pre-specify an empty kernel matrix row sum to update as we go
+        kernel_row_sum = jnp.zeros(len(x))
+
+        # Define the kernel object
+        kernel = coreax.kernel.SquaredExponentialKernel(length_scale=length_scale)
```

Comment:
Is there a reason we're using a specific concrete implementation of `Kernel` here, rather than just using `patch.multiple` to allow instantiating `Kernel` directly? (Or, equally, defining a dummy `Kernel` subclass with trivial implementations of the required abstract methods).

Reply:
I did consider using a new class with trivial implementations of each method, but in hindsight felt that just using the most standard, basic real kernel was actually easier to read and follow

Reply:
I think in the spirit of properly isolated unit testing we _should_ be using some kind of dummy/patched implementation, but as I said it's not an urgent change.